instrumentation/pulsar/pulsar-2.8/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/PulsarClientSuppressReceiveSpansTest.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pulsar.v2_8;
+
+import static io.opentelemetry.instrumentation.testing.util.TelemetryDataUtil.orderByRootSpanName;
+
+import io.opentelemetry.api.trace.SpanKind;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.MessageId;
+import org.apache.pulsar.client.api.MessageListener;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SubscriptionInitialPosition;
+import org.junit.jupiter.api.Test;
+
+class PulsarClientSuppressReceiveSpansTest extends AbstractPulsarClientTest {
+
+  @Test
+  void testSendNonPartitionedTopic() throws Exception {
+    String topic = "persistent://public/default/testSendNonPartitionedTopic";
+    admin.topics().createNonPartitionedTopic(topic);
+    producer = client.newProducer(Schema.STRING).topic(topic).enableBatching(false).create();
+
+    String msg = "test";
+    MessageId msgId = testing.runWithSpan("parent", () -> producer.send(msg));
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName(topic + " publish")
+                        .hasKind(SpanKind.PRODUCER)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            sendAttributes(topic, msgId.toString(), false))));
+  }
```

Comment:
This kinda duplicates the first half of the client test below, since we have to send something in order to to the client receive. With that, do you still think it's necessary to keep this test on its own? Or is the behavior (also asserted) in the client test sufficient?

Reply:
Although we could move this test to `AbstractPulsarClientTest` to get rid of duplication I'll delete `testSendNonPartitionedTopic` and `testSendPartitionedTopic`. There are also consumer tests that contain the exact same producer code which already verify this behavior.